crates/ra_syntax/src/grammar/types.rs

```diff
@@ -243,6 +243,28 @@ pub(super) fn path_type(p: &mut Parser) {
     path_type_(p, true)
 }
 
+// test macro_call_type
+// type A = foo!();
+// type B = crate::foo!();
+fn path_or_macro_type_(p: &mut Parser, allow_bounds: bool) {
+    assert!(paths::is_path_start(p) || p.at(L_ANGLE));
+    let m = p.start();
+    paths::type_path(p);
```

Comment:
This accepts paths with `Mode::Type` for path types and macro calls, but macro calls should really have `Mode::Use` AFAIK.